src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1577,19 +1554,14 @@ protected override void ProcessRecord()
         /// Because this function sets the request's ContentLength property and writes content data into the requests's stream,
         /// it should be called one time maximum on a given request.
         /// </remarks>
-        internal long SetRequestContent(HttpRequestMessage request, byte[] content)
+        internal ByteArrayContent SetRequestContent(HttpRequestMessage request, byte[] content)
         {
             ArgumentNullException.ThrowIfNull(request);
+            ArgumentNullException.ThrowIfNull(content);
 
-            if (content is null)
-            {
-                return 0;
-            }
-
-            var byteArrayContent = new ByteArrayContent(content);
-            request.Content = byteArrayContent;
+            ByteArrayContent byteArrayContent = new(content);
 
-            return byteArrayContent.Headers.ContentLength.Value;
```

Comment:
As I pointed above, I'd prefer to follow the method name - leave line 1590 and remove return.